### ISSUE TITLE
Tileset editing leads to unrecoverable crash

### ISSUE THREAD

**Issue by jgestiot** [godotengine/godot#74422]
### Godot version

Godot Engine v4.0.rc6.official.0cd148313

### System information

Windows 11 Home 10.0.22621 Build 22621

### Issue description

I was setting up a TileMap/TileSet  when the entire application crashed. It crashed twice in a row . Each time it happened, I was resizing the height of the Tileset editing area using the mouse. Godot hangs as soon as I resize the height of the Tileset editor. The hour glass is displayed and it hangs indefinitely. The only option is to get Windows to close it. When I reload my project, everything related to the TileMap has disappeared including the TileMap node itself.

### Steps to reproduce

- Create a scene
- Create a TileMap node
- Create an Atlas
- Create a physics then a navigation then a custom data layers
- Resize the height of the TileSet area
- Godot hangs. It happened to me twice in a row. 
- All data worked on the TileMap is gone

### Minimal reproduction project

I do not have a project because the issue wipes out whatever I did in the project. I did a screeshot of Godot hanging.
![Screenshot (86)](https://user-images.githubusercontent.com/6059364/222950069-b93e9a7c-ef6b-4330-a9bc-6e0030b14ece.png)

**Comment by a-wats** [comment]
I've been running into similar issues but can't find a pattern to them. I've been working on a TileSet attached to a TileMap node, in a Scene created with godot 4 beta. Since upgrading to the latest stable release, It's crashed many times:

1. Making the TileMap node invisible.
2. Trying to place tiles in the scene
3. Opening the Scene that contains this TileMap node.
4. Selecting the TileMap node in the Scene Tree.
5. Editing Terrain data

Selecting the TileMap node has caused Godot to crash 6+ times now, but not consistently. The rest have been single instances.

**Comment by jgestiot** [comment]
I have also had a crash clicking into the Cell Quadrant Size to change the value from the default 16. As soon as I clicked in, the whole thing hanged as I described in my post.

**Comment by DSPerson** [comment]
version: 4.0 
os: M1 macOS 12.6
100% crash when I edit TileMap terrain name.
<img width="385" alt="image" src="https://user-images.githubusercontent.com/17868782/223003494-8b4e9bef-6d8f-40d7-8f39-413fcf38fce1.png">

**Comment by jgestiot** [comment]
I also get the same crash with the isometric project at https://github.com/godotengine/godot-demo-projects

**Comment by nbourre** [comment]
Using Godot 4.0.2 under Windows 11, I can easily reproduce the problem.

- Add a Tilemap and a new Tileset.
- Create a new Atlas
- Quick Load a PNG
- Click into the inspector of the Atlas
- It crashes

Here's a screen recording showing the problem.

https://user-images.githubusercontent.com/2332679/236465973-b539c2e1-aa9e-419c-99fe-11fb79468066.mp4

**Comment by KoBeWi** [comment]
Can't reproduce with the same steps. Is this an empty project? Can you share the Tilesheet? Maybe it's related.

**Comment by nbourre** [comment]
> Does this still happen on latest version ? i could not reproduce the bug.

Yes it still does crash. I attached the tilesheet to this message.

![Tilesheet](https://github.com/godotengine/godot/assets/2332679/6dbad640-2533-4082-b4bd-1ba8014fd4c5)

**Comment by groud** [comment]
> Yes it still does crash. I attached the tilesheet to this message.

According to your video, that's kind of expected. You are creating around 35000 tiles by running the automatic tile creation algorithm. Your texture is 3456 x 2560 for a tile size of 16 x 16! So I'm not even sure it's a crash, the editor simply hangs processind the things.

I am not sure if reaching the limitation of your machine is really a bug TBH. Though I guess that we should probably add a warning when you are about to automatically create too many tiles like that.

**Comment by YuriSizov** [comment]
Sounds like it's a bit of an edge case. Editor UX definitely needs improving from what I'm seeing, but it's not blocking 4.1.

**Comment by akien-mga** [comment]
I don't seem to reproduce the issue in the OP in 4.4.1.stable, so I assume it has been fixed (probably a while ago).
